### ISSUE TITLE
UnsatisfiedLinkError: No implementation found for void com.mapbox.mapboxsdk.net.NativeConnectivityListener.initialize()

### ISSUE THREAD

**Issue by arbitur** [mapbox/mapbox-gl-native#15535]
There seems to be an intermittent issue with Mapbox 8.2.1 (I updated from 5.5 to 8.2). I get an `UnsatisfiedLinkError No implementation found for void com.mapbox.mapboxsdk.net.NativeConnectivityListener.initialize()` on some of my customers devices. The crash happens on several different Android version (Observed 6.x.x-9.x.x) and different hardware. The only common characteristic is that the devices are rooted.

Any ideas?

**Comment by LukasPaczos** [comment]
Hey @arbitur, when are you calling `Mapbox#getInstance`? Also, you could experiment with different native library loaders like [`ReLinker`](https://github.com/KeepSafe/ReLinker) using the `LibraryLoader#setLibraryLoader`. Maps SDK uses the [`SoLoader`](https://github.com/facebook/SoLoader) by default.

**Comment by arbitur** [comment]
@LukasPaczos Hi, Im calling it in Application.java::onCreate:

```java
public final class MainApplication extends Application {
    @Override
        public void onCreate() {
            super.onCreate();
            Mapbox.getInstance(this, getString(R.string.mapbox_access_token));
        }
}
```

**Comment by LukasPaczos** [comment]
Thanks @arbitur, this looks good. In this case, it looks like the library loader is failing. Are you getting any logs with `Mbgl-LibraryLoader` tag?

Please feel free to experiment with the library loaders linked above, or even reverting back to the system one (`System#loadLibrary`).

Closing as this is not actionable from the Maps SDK perspective, this is a native library loader or Android OS issue.

**Comment by arbitur** [comment]
@LukasPaczos But Mapbox are the ones using a potentially flawed loader so this is very much actionable from the Mapbox perspective.

**Comment by LukasPaczos** [comment]
@arbitur We've iterated through all system, ReLinker and SoLoader libraries and SoLoader has proven to be the most reliable one to date. Unfortunately, there's not much more we can do at this point, with the limited information and reports we have.

**Comment by arbitur** [comment]
@LukasPaczos Which loader did you use in v5? This issue started once I updated to v8.

**Comment by LukasPaczos** [comment]
Looks like it was a `System` one - https://github.com/mapbox/mapbox-gl-native/blob/release-agua/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/LibraryLoader.java#L15.

**Comment by philemonmerlet** [comment]
Hello, fyi I opened a similar issue, with "steps" to reproduce the crash.

**Comment by bc67da8d** [comment]
This issue is duplicate and I solved it in #14428
